packages/suite/src/middlewares/suite/analyticsMiddleware.ts

```diff
@@ -115,71 +107,32 @@ const analytics =
                         return acc;
                     }, {});
 
-                api.dispatch(
-                    analyticsActions.report({
-                        type: 'accounts/status',
-                        payload: { ...accountsStatus },
-                    }),
-                );
+                analytics.report({
+                    type: 'accounts/status',
+                    payload: { ...accountsStatus },
+                });
                 break;
             }
             case ROUTER.LOCATION_CHANGE:
-                api.dispatch(
-                    analyticsActions.report({
-                        type: 'router/location-change',
-                        payload: {
-                            prevRouterUrl,
-                            nextRouterUrl: action.payload.url,
-                        },
-                    }),
-                );
+                analytics.report({
+                    type: 'router/location-change',
+                    payload: {
+                        prevRouterUrl,
+                        nextRouterUrl: action.payload.url,
+                    },
+                });
                 break;
             case SUITE.AUTH_DEVICE:
-                api.dispatch(
-                    analyticsActions.report({
-                        type: 'select-wallet-type',
-                        payload: { type: action.payload.walletNumber ? 'hidden' : 'standard' },
-                    }),
-                );
-                break;
-            case SUITE.TOR_STATUS:
```

Comment:
just out of curiosity. why did you feel it is better to move reporting from here to other places?

Reply:
Tor event was sent on each desktop app launch. It should be sent only on change.

Regarding `BACKUP.SET_STATUS` and `BACKUP.SET_ERROR`, I think that in this case it is better to have it on a place where those actions are fired.